### PR TITLE
Support CloudEvents type cleanup while publishing

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Add the support for CloudEvent type cleanup while publishing.

This will keep Kyma Eventing user experience consistent by subscribing and publishing to the same event types.

Given the following Kyma Subscription:

```yaml
apiVersion: eventing.kyma-project.io/v1alpha1
kind: Subscription
metadata:
  name: test-subscription
  namespace: default
spec:
  filter:
    dialect: beb
    filters:
      - eventSource:
          property: source
          type: exact
          value: ""
        eventType:
          property: type
          type: exact
          value: sap.kyma.custom.no-app.Some-Event-Name.Some-Event-Type.v1
  protocol: ""
  protocolsettings:
    exemptHandshake: true
    qos: AT-LEAST-ONCE
  sink: http://subscriber.default.svc.cluster.local
```

Event publishers can publish CloudEvents with type `sap.kyma.custom.no-app.Some-Event-Name.Some-Event-Type.v1`.

### CODE REVIEW

File: components/event-publisher-proxy/pkg/legacy-events/legacy_test.go [hunk 1]
```diff
@@ -177,17 +182,17 @@ func TestConvertPublishRequestToCloudEvent(t *testing.T) {
...
 	if gotEvent.Context.GetTime() != timeNowFormatted {
 		t.Errorf("incorrect eventTime, want: %s, got: %s", timeNowFormatted, gotEvent.Context.GetTime())
 	}
 	if gotEvent.Context.GetSource() != bebNs {
 		t.Errorf("incorrect source want: %s, got: %s", bebNs, gotEvent.Context.GetDataContentType())
 	}
-	if gotEvent.Context.GetDataContentType() != ContentTypeApplicationJSON {
-		t.Errorf("incorrect content-type, want: %s, got: %s", ContentTypeApplicationJSON, gotEvent.Context.GetDataContentType())
+	if gotEvent.Context.GetDataContentType() != internal.ContentTypeApplicationJSON {
+		t.Errorf("incorrect content-type, want: %s, got: %s", internal.ContentTypeApplicationJSON, gotEvent.Context.GetDataContentType())
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/legacy-events/legacy_test.go]: Done.

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**